sqlalchemy_hana/dialect.py

```diff
@@ -239,7 +239,11 @@ def _get_server_version_info(self, connection):
         pass
 
     def _get_default_schema_name(self, connection):
-        return self.normalize_name(connection.engine.url.username.upper())
+        # return self.normalize_name(connection.engine.url.username.upper())
+        with connection.cursor() as cursor:
+            cursor.execute("SELECT CURRENT_USER FROM DUMMY")
+            result = cursor.fetchone()
+        return result[0]
```

Comment:
The return value should be the normalized schema name. In your new implementation it will be a username which is normally UPPER case. Calling `self.normalize_name` should work.

Reply:
code change and now included in the PR